src/App.tsx

```diff
@@ -1,39 +1,73 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
+
+type Props = {};
+
+let timerId: number | null = null;
+
+export class App extends React.Component<Props, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount() {
+    this.makeRenameInterval();
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.addEventListener('contextmenu', (event: MouseEvent) => {
+      event.preventDefault();
+      this.setState((previousState) => ({ ...previousState, hasClock: false }));
+      window.clearInterval(timerId as number);
+    });
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    document.addEventListener('click', (event: MouseEvent) => {
+      event.preventDefault();
+      if (!this.state.hasClock) {
+        this.setState((previousState) => ({
+          ...previousState,
+          hasClock: true,
+        }));
+        this.makeRenameInterval();
+      }
+    });
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  componentWillUnmount() {
+    document.removeEventListener('contextmenu', () => { });
+    document.removeEventListener('click', () => { });
+  }
 
-        {' time is '}
+  makeRenameInterval() {
+    clearInterval(timerId as number);
+    timerId = window.setInterval(() => {
+      const clockName = getRandomName();
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+      /* eslint-disable no-console */
+      console.debug(`Renamed from ${this.state.clockName} to ${clockName}`);
```

Comment:
move this logic to componentDidUpdate
and change to `Renamed from ${prevProps.clockName} to ${this.props.clockName}`

Reply:
done, but I move it to Clock